### PR TITLE
feat(attest): reject votes for unsupported conf levels

### PR DESCRIPTION
**Pull request by arajasek**
This ensures that we never create or verify votes for unsupported Conf levels.

issue: #1579

### CODE REVIEW

**Comment by arajasek** [comment]
Tests are failing because we're creating votes for `Finalized` for the chain `IDMockL1Fast` even though [the only supported shard](https://github.com/omni-network/omni/blob/897d2db4fdd3b2053d5cfd345aea83ca2eea9d16/lib/netconf/static.go#L225) for that chain for the test is `Latest`.

I'm not yet sure whether this is just a test thing and we should simply add `Finalized` as a supported level, or whether it's actually a bug that we're creating votes for `Finalized` when it's not technically supported.

File: halo/registry/types/interface.go [hunk 1]
```diff
@@ -10,6 +10,6 @@ type PortalRegistry interface {
 	// SupportedChain returns true if the chain is included in the registry.
 	SupportedChain(ctx context.Context, chainID uint64) (bool, error)
```
Comment (corverroos) [inline halo/registry/types/interface.go:11]: Why don't we clean this up while we are here:
 - Remove `SupportedChain` from interface, implementation and tests
 - Rename `ConfLevels` to something like `SupportedChainVersions`
 - Refactor all uses of `SupportedChain` to use `SupportedChainVersions`.
 - Ensure we have unit tests for `SupportedChainVersions`.

File: halo/attest/keeper/keeper.go [hunk 2]
```diff
@@ -631,11 +631,8 @@ func (k *Keeper) ExtendVote(ctx sdk.Context, _ *abci.RequestExtendVote) (*abci.R
 	duplicate := make(map[types.VoteSource]bool)
 	var filtered []*types.Vote
 	for _, vote := range votes {
-		if ok, err := k.portalRegistry.SupportedChain(ctx, vote.BlockHeader.ChainId); err != nil {
-			return nil, errors.Wrap(err, "supported chain")
-		} else if !ok {
-			log.Warn(ctx, "Skipping own vote for unsupported chain", nil, "chain", k.namer(vote.BlockHeader.XChainVersion()))
```
[the remark this answers is not in the document]
Comment (corverroos) [reply halo/attest/keeper/keeper.go:637]: cool

**Comment by corverroos** [comment]
wrt unit tests, yeah, the data used in tests is in most cases not perfect, it is generally just "good enough" for the tests to pass. When adding new logic and validation, tests may fail. The data then needs to be improved to aligned with new logic and validation.

**Review by corverroos — APPROVED** [review]

**Review by corverroos — APPROVED** [review]

**Pull request merged**